src/WebJobs.Script/Diagnostics/HostFileLoggerProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides a logger for writing all host logs to a single Host file.
+    /// </summary>
+    internal class HostFileLoggerProvider : ILoggerProvider
```

Comment:
Why don't we just have a single logger provider for files, that just takes the path to discriminate? Host v.s. Function logs differs by path only, right?

Reply:
They're the same except that every log from the Host goes to the Host logs, but the logs to the functions are filtered. In other words, a single log message can go to either the Host -- or the Host + Function.

So I could have one file logger that logs everything to the Host file and then filters based on whether it's a file logger or not. But that complicated things (I'd need a single writer to the Host file, due to our locks for example). In order to manage it better and separate the concerns, I kept it separate. This is similar to how the TraceWriter works today -- we create different ones based on the files.